fdbserver/SkipList.cpp

```diff
@@ -379,16 +379,9 @@ class SkipList : NonCopyable
 	};
 
 	static force_inline bool less( const uint8_t* a, int aLen, const uint8_t* b, int bLen ) {
-		int len = min(aLen, bLen);
-		for(int i=0; i<len; i++)
-			if (a[i] < b[i])
-				return true;
-			else if (a[i] > b[i])
-				return false;
-
-		/*int c = memcmp(a,b,min(aLen,bLen));
+		int c = memcmp(a,b,min(aLen,bLen));
 		if (c<0) return true;
-		if (c>0) return false;*/
+		if (c>0) return false;
```

Comment:
It's interesting to me that this was written the `memcmp` way initially and then changed. I don't see any reason why it needs to be like it is, and the 10 year old commit that changed it doesn't make it any clearer to me.

I'll run it through some simulated tests to make sure nothing pops up.

Reply:
Seemed ok in testing.